### PR TITLE
Different scan map pr

### PR DESCRIPTION
**Pull request by dillonfranke**

### CODE REVIEW

File: go/src/pcap/pcap.go [hunk 1]
```diff
@@ -120,47 +116,95 @@ var zMapMap map[uint16]map[int]int
 
 var masscanMap map[uint16]map[int]int
 
+var scanMap map[uint16]map[uint16]int
+var firstPacketTime map[uint16]time.Time
+var recentPacketTime map[uint16]time.Time
```
Comment (dillonfranke) [inline go/src/pcap/pcap.go:120]: Assuming this maps IP src to times for first and recent packets. Good idea.

File: go/src/pcap/pcap.go [hunk 2]
```diff
@@ -120,47 +116,95 @@ var zMapMap map[uint16]map[int]int
...
 
-	return difference < goalDuration
+	longDifference := int(recent.Sub(oldestPacket))
+	numPackets := 0
+	scanMut.Lock()
+	for _, v := range allDests {
+		numPackets += v
+	}
+	scanMut.Unlock()
+	average := float64(numPackets) / (float64(longDifference) / float64(10e9))
```
Comment (dillonfranke) [inline go/src/pcap/pcap.go:162]: What exactly is this calculating, unit-wise? the cutoff is 0.1, but what is that units for?

File: go/src/pcap/pcap.go [hunk 3]
```diff
@@ -195,72 +238,63 @@ func handlePackets(filename string, wg *sync.WaitGroup) {
...
+			//fmt.Printf("Source IP: %s\n", ip.SrcIP)
+			//fmt.Printf("Destin IP: %s\n", ip.DstIP)
+			//fmt.Printf("Protocol: %s\n", ip.Protocol)
+			//ipId = ip.Id
+			ipSrc = ip.SrcIP
+			ipDest = ip.DstIP
+
 		} else {
-			fmt.Println("Skipping...")
+			fmt.Println("I didn't want this packet anyways")
```
Comment (dillonfranke) [inline go/src/pcap/pcap.go:277]: same

**Review by dillonfranke — COMMENTED** [review]
Nice this makes sense to me. Looks good.

File: go/src/pcap/pcap.go (hunk 3 above)
Comment (wkautz) [reply go/src/pcap/pcap.go:277]: In 240 million packets there were like 400 that got skipped

File: go/src/pcap/pcap.go (hunk 2 above)
Comment (wkautz) [reply go/src/pcap/pcap.go:162]: that is packets per second. So numPackets is number of packets. longDifference is time difference in nanoseconds, so i divide it by 10e9

File: go/src/pcap/pcap.go [hunk 4]
```diff
@@ -120,47 +113,101 @@ var zMapMap map[uint16]map[int]int
...
+		}
+		if totalPackets >= SCAN_CUTOFF {
+			scansSizes[ipSrc] = append(scansSizes[ipSrc], totalPackets)
+
+			var scanPorts map[uint16][][]uint32
+			var allPorts []uint32
+			for k := range scanPortMap[ipSrc] {
+				allPorts = append(allPorts, k)
+			}
+			scanPorts[ipSrc] = append(scanPorts[ipSrc], allPorts)
```
Comment (dillonfranke) [inline go/src/pcap/pcap.go:183]: This is where the panic is occurring.

**Pull request merged**